x-pack/metricbeat/module/gcp/carbon/carbon.go

```diff
@@ -0,0 +1,320 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package carbon
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/civil"
```

Comment:
For some reason, golangci-lint considers this is not being used, which is not accurate. It's used at line 280.

https://github.com/elastic/beats/pull/34820/files#diff-264deb4dfc253bbf225e6d4331bf1111e284c2066e027e6df36ee0674d9eb1c7R280

Reply:
yah, `typecheck` has a number of aggravating edge cases, and using the import just a for a type cast might be one of them. Assuming it's not to much trouble, it might be worth experimenting and using a function or something from `civil` and seeing if it makes the linter happy. Can also just use a `//nolint:typecheck //comment` comment, but I've had issues with those not working for `typecheck`.

Reply:
We get a [Value](https://pkg.go.dev/cloud.google.com/go/bigquery#Value), which is a plain `interface{}`.
So no, unfortunately, without any type casting, we can't call any method.